cura/PrinterOutputDevice.py

```diff
@@ -0,0 +1,303 @@
+    def hotendTemperatures(self):
+        return self._hotend_temperatures
+
+    ##  Protected setter for the current hotend temperature.
+    #   This simply sets the hotend temperature, but ensures that a signal is emitted.
+    #   /param index Index of the hotend
+    #   /param temperature temperature of the hotend (in deg C)
+    def _setHotendTemperature(self, index, temperature):
+        self._hotend_temperatures[index] = temperature
+        self.hotendTemperaturesChanged.emit()
+
+    ##  Attempt to establish connection
+    def connect(self):
+        pass
+
+    ##  Attempt to close the connection
+    def close(self):
+        pass
+
+    @pyqtProperty(bool, notify = connectionStateChanged)
+    def connectionState(self):
+        return self._connection_state
+
+    ##  Set the connection state of this output device.
+    #   /param connection_state ConnectionState enum.
+    def setConnectionState(self, connection_state):
+        self._connection_state = connection_state
+        self.connectionStateChanged.emit(self._id)
+
+    ##  Ensure that close gets called when object is destroyed
+    def __del__(self):
+        self.close()
+
+    ##  Get the x position of the head.
+    #   This function is "final" (do not re-implement)
+    @pyqtProperty(float, notify = headPositionChanged)
+    def headX(self):
+        return self._head_x
+
+    ##  Get the y position of the head.
+    #   This function is "final" (do not re-implement)
+    @pyqtProperty(float, notify = headPositionChanged)
+    def headY(self):
+        return self._head_y
+
+    ##  Get the z position of the head.
+    #   In some machines it's actually the bed that moves. For convenience sake we simply see it all as head movements.
+    #   This function is "final" (do not re-implement)
+    @pyqtProperty(float, notify = headPositionChanged)
+    def headZ(self):
+        return self._head_z
+
+    ##  Set the position of the head.
+    #   In some machines it's actually the bed that moves. For convenience sake we simply see it all as head movements.
+    #   This function is "final" (do not re-implement)
+    #   /param x new x location of the head.
+    #   /param y new y location of the head.
+    #   /param z new z location of the head.
+    #   /param speed Speed by which it needs to move (in mm/minute)
+    #   /sa _setHeadPosition implementation function
+    @pyqtSlot("long", "long", "long")
+    @pyqtSlot("long", "long", "long", "long")
+    def setHeadPosition(self, x, y, z, speed = 3000):
+        self._setHeadPosition(x, y , z, speed)
+
+    ##  Set the X position of the head.
+    #   This function is "final" (do not re-implement)
+    #   /param x x position head needs to move to.
+    #   /param speed Speed by which it needs to move (in mm/minute)
+    #   /sa _setHeadx implementation function
+    @pyqtSlot("long")
+    @pyqtSlot("long", "long")
+    def setHeadX(self, x, speed = 3000):
+        self._setHeadX(x, speed)
```

Comment:
I see self._head_y as a stored value (posibiliy to be updated by the listen thread or some other means). Setting the headPosition should use new values (instead of stored ones).